test/unit/mock/fake_runtime_service.py

```diff
@@ -82,20 +100,28 @@ def _initialize_hgps(
                 client_params=hgp_params, instance=hgp_name, service=self
             )
             fake_account_client = self._fake_account_client
-            if not fake_account_client:
-                specs = [
-                    {"configuration": {"backend_name": self.DEFAULT_COMMON_BACKEND}},
-                    {
-                        "configuration": {
-                            "backend_name": self.DEFAULT_UNIQUE_BACKEND_PREFIX
-                            + str(idx)
-                        }
-                    },
-                ]
-                fake_account_client = BaseFakeAccountClient(specs=specs, hgp=hgp_name)
+            specs = [
+                {"configuration": {"backend_name": self.DEFAULT_COMMON_BACKEND}},
+                {
+                    "configuration": {
+                        "backend_name": self.DEFAULT_UNIQUE_BACKEND_PREFIX + str(idx)
+                    }
+                },
+            ]
+            fake_account_client = BaseFakeAccountClient(specs=specs, hgp=hgp_name)
             hgp._runtime_client = fake_account_client
+            self._fake_account_client = fake_account_client
+            for backend_name, backend in hgp.backends.items():
+                if not backend:
+                    self._set_backend_config(backend_name, hgp_name)
+                    backend_obj = ibm_backend.IBMBackend(
+                        instance=hgp_name,
+                        configuration=self._backend_configs[backend_name],
+                        service=self,
+                        api_client=hgp._runtime_client,
+                    )
+                    hgp.backends[backend_name] = backend_obj
```

Comment:
Not sure what the best way to update these unit tests is without significant refactoring - thoughts? @jyu00

Reply:
From what I can tell this `_initialize_hgps()` method doesn't actually collect the backend configuration. It just initializes a fake account client, which will need to be changed to fake runtime client, since that's what is used now. The fake runtime client will also need to be updated to take the "spec", to know what backends to return.

Then in theory when the mainline code calls `self._api_client.backend_configuration`, it should go to the FakeRuntimeClient to get the backend config.

Reply:
I believe the hgps on the `FakeRuntimeService` side will still need the backend configs somehow because when jobs are retrieved, the service will initialize without any backends 

then `self._api_client.backend_configuration` would result in in a backend cannot be found error